cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/AppLifecycleHttpHandler.java

```diff
@@ -229,23 +221,16 @@ public void deleteApp(HttpRequest request, HttpResponder responder,
   @DELETE
   @Path("/apps")
   public void deleteAllApps(HttpRequest request, HttpResponder responder,
-                            @PathParam("namespace-id") String namespaceId)
-    throws BadRequestException, NamespaceNotFoundException {
-
+                            @PathParam("namespace-id") String namespaceId) throws Exception {
     Id.Namespace id = validateNamespace(namespaceId);
-    AppFabricServiceStatus status;
     try {
       applicationLifecycleService.removeAll(id);
-      status = AppFabricServiceStatus.OK;
-    } catch (NotFoundException nfe) {
-      status = AppFabricServiceStatus.PROGRAM_NOT_FOUND;
-    } catch (CannotBeDeletedException cbde) {
-      status = AppFabricServiceStatus.PROGRAM_STILL_RUNNING;
-    } catch (Exception e) {
-      status = AppFabricServiceStatus.INTERNAL_ERROR;
+      LOG.trace("Delete all call at AppFabricHttpHandler");
+      responder.sendStatus(HttpResponseStatus.OK);
+    } catch (CannotBeDeletedException e) {
+      // Keeping this for backward compatibility. Ideally this should return conflict, not forbidden.
+      responder.sendString(HttpResponseStatus.FORBIDDEN, "Program is still running");
```

Comment:
Its currently difficult to know without changing the ``ProgramRuntimeService.checkAnyRunnning()`` method. Currently, that only returns ``true`` or ``false``. Reported https://issues.cask.co/browse/CDAP-5415 to fix this later.